[READY] DRAKEBORGS & PLUSHIES

Strazyplus opened the pull request:
### **[Attribution-NonCommercial-ShareAlike 3.0 Unported (CC BY-NC-SA 3.0)](https://creativecommons.org/licenses/by-nc-sa/3.0/)**

![1595361337 miz-dee_okay-export](https://user-images.githubusercontent.com/6706750/88109990-70aba580-cb79-11ea-8a4d-2418b4207298.gif)

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Ported from skyrat https://github.com/Skyrat-SS13/Skyrat13/pull/2863
Commissioned artist: **[deviantart.com/mizartz](deviantart.com/mizartz)**
Vorebellies: @ThighHighs

Though a commission with the talented Mizartz we have created a new subset of borgs for all to have, instead of being a dog the robot is a dragon or in this case a drake. the modules include Medical, Security, Mining, Peace, Engineering, and Janitor variants. in addition of  providing new borgshells we have also included plushie variants that can be obtained via plushie crate or via loadout and come in security and medical. 

![janitor](https://user-images.githubusercontent.com/6706750/88110509-56be9280-cb7a-11ea-9520-0e712ed6f325.gif)
![mining](https://user-images.githubusercontent.com/6706750/88110511-57572900-cb7a-11ea-9fab-29fee8b2f8dd.gif)
![peace](https://user-images.githubusercontent.com/6706750/88110512-57572900-cb7a-11ea-9c89-95c99773f98a.gif)
![security](https://user-images.githubusercontent.com/6706750/88110513-57572900-cb7a-11ea-8132-fa5ddfbc5c89.gif)
![engineering](https://user-images.githubusercontent.com/6706750/88110515-57efbf80-cb7a-11ea-96ad-00c43b64905a.gif)
![medic](https://user-images.githubusercontent.com/6706750/88110516-57efbf80-cb7a-11ea-8704-ca08dc83f998.gif)
![ca2rd](https://user-images.githubusercontent.com/6706750/87237091-b4572000-c3bf-11ea-904e-b42853267d4e.png)

Drakeborg and drake plushies fall under [CC BY-NC-SA 3.0](https://creativecommons.org/licenses/by-nc-sa/3.0/)
Share — copy and redistribute the material in any medium or format, Adapt — remix, transform, and build upon the material, however NonCommercial use — You may not use the material for commercial purposes.

this pr closes #12735 the pr of [Blackmajor's Drake Borg port](https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12735).

## Why It's Good For The Game

Brings new subset of robots for borg players as well cute plushies for people to squeeze.

## Changelog
:cl:
add: Added drakeborgs
add: Added drakeplushies
imageadd: added drakeborg sprites
imageadd: added drakeplushie sprites 
code: changed some code - added drakeplushies to backpack loadout
Removed duplicate voresleeper belly sprites from engdrake & jantidrake.
[CC BY-NC-SA 3.0](https://creativecommons.org/licenses/by-nc-sa/3.0/)
add: Added CC BY-NC-SA 3.0 license details to icon/mob/cyborg
moved drakeborg.dmi to icon/mob/cyborg
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

BlackMajor committed:
bet?

BlackMajor committed:
done

Strazyplus committed:
Merge branch 'pr/12735' into DRAKEBORGS

Strazyplus committed:
Dragon online!

Strazyplus committed:
Caution crushing hazard

Strazyplus committed:
Who feeding the chunkydrakes

BlackMajor commented on code/modules/mob/living/silicon/robot/robot_modules.dm:
```diff
@@ -428,6 +428,7 @@
 			cyborg_base_icon = "drakemed"
 			cyborg_icon_override = 'modular_citadel/icons/mob/drakemech.dmi'
 			sleeper_overlay = "drakemedsleeper"
+			sleeper_overlay_override = 'modular_citadel/icons/mob/drakemech.dmi'
```
```suggestion
```
This line is unnecessary.

Strazyplus replied:
oops that slipped

Strazyplus committed:
Removed unused sleeper_override

silicons reviewed (changes requested):
everything is still pending considering bhijn is yet to hear back from the artist
regardless, i am in favor of drakeborgs being in a separate file as they are a new class of cyborg sprites. however, they should be in a cyborg icon directory or something, maybe make icons/mob/cyborg/drakes.dmi or something.
there's also another reason to do this as i believe the precedent for differently licensed files on tgcode was to mark the file with licensing information in the directory and separating it out would make this easily able to happen.

credit wise: there is no reason to put the license and everything on 7 places in the code.
put all of this information in a license .txt marking the file you put the sprites in on the folder. maybe a comment at the top of robot_modules.dm saying that drake sprites have different requirements/licensing than the rest of the assets would be acceptable. 

i personally also feel like it would be right to allow navier right of way for this pull request considering they were the original author assuming they are willing to compromise via the above. and of course, if you have any issues with this, talk to me on discord on coderbus @Strazyplus

Strazyplus committed:
Lawyer-drake when?

unmodularized drakeborg
added licencing documentation
reduced / simpified code comments

Strazyplus committed:
More License detail tweaks.

LetterN commented on code/game/objects/items/license.txt:
```diff
@@ -0,0 +1,3 @@
+icons/obj/plushies.dmi's icon state of secdrake and meddrake by Mizartz. It has been licensed under the CC BY-NC-SA 3.0 license.
```
place this into the icons/ instead of code/game/objects/items

Strazyplus replied:
Moved to icons/

LetterN commented on code/game/objects/items/plushes.dm:
```diff
@@ -1,3 +1,5 @@
+//Drakeplushies are licensed under the CC BY-NC-SA 3.0 license.
+
```
delete this, they're already atributed on the license.txt for the _icons_, as they made the icon.

LetterN commented on code/modules/mob/living/silicon/robot/robot_modules.dm:
```diff
@@ -1,3 +1,5 @@
+// Drakeborg sprites by Mizartz. drakeborg are licensed under the CC BY-NC-SA 3.0 license. licenseing infomation located in \icons\mob\cyborg
+
```
delete this

Strazyplus replied:
took silicons suggestion by instead of seven comments just make one mentioning the credit & the licence on the top.

LetterN commented on code/game/objects/items/license.txt:
```diff
@@ -0,0 +1,3 @@
+icons/obj/plushies.dmi's icon state of secdrake and meddrake by Mizartz. It has been licensed under the CC BY-NC-SA 3.0 license.
+
```
```suggestion
icons/obj/plushies.dmi's icon state of secdrake and meddrake by Mizartz. It has been licensed under the CC BY-NC-SA 3.0 license.
icons/mob/cyborg/drakemech.dmi by Mizartz. It has been licensed under the CC BY-NC-SA 3.0 license.
```
recommend adding that too

Strazyplus replied:
late, will edit morning + was silicons suggestion for comment on top.

silicons reviewed (dismissed):
acceptable but get rid of code/game/obejcts/items/license.txt as the license isn't for the code it's for the art

**Do not take this approval as my support for the *content* of the PR, I am simply okay with the status of the licensing/attribution at this point.** I don't have any personal preference on the former.

Strazyplus committed:
Drakes are dragons too

Strazyplus committed:
tweak of drakemechs that you will never notice

timothyteakettle commented on modular_citadel/code/modules/client/loadout/backpack.dm:
```diff
@@ -23,6 +23,16 @@
+	name = "MediDrake Plushie"
+	category = SLOT_IN_BACKPACK
+	path = /obj/item/toy/plush/borgplushie/meddrake
+
```
personally its better to scrap loadout plushies and make it a choice beacon so people can start with /any/ loadout plushie (apart from those with special effects)

edit: fuck it I can just do this myself but uh yeah, loadout bloat bad, consider removing them

Strazyplus replied:
I am a terrible coder, I have never messed with choice beacon code, while that would reduce bloat on loadout list this would, however, prevent people from having the choice of having multiple plushies if that is what they wanted. the plushies are added so they can be obtained.

Strazyplus committed:
Don't DragON me, mister!

Removes comments from code
Removes mentions of Navier
Why are we still here? Just to suffer?

deathride58 reviewed (dismissed):
UPDATE: I have spoke with the artist! They have given express consent for the content here to be licensed under CC-BY-NC-SA 3.0, with the NC clause being under the interpretation that commercial use constitutes monetizing the materials directly (IE, making them a part of donator-exclusive features, or otherwise locking them behind a paywall). This means that they're *teeeeeechnically* GPL-compatible, and also means that it's compatible with the way we handle donations (we don't have to worry about T2 or T3 donations being a thing, since these items/skins are available for everyone)

Strazyplus committed:
Merge conflict because of a hairball? mmkay

actions-user committed:
Automatic changelog compile [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

Strazyplus committed:
Merge branch 'DRAKEBORGS' into DRAKEBORG

Strazyplus committed:
Stupid conflicts coming from closed prs how nice?

Strazyplus committed:
Merge branch 'master' into DRAKEBORG

actions-user committed:
Automatic changelog compile [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

actions-user committed:
Automatic changelog compile [ci skip]

Strazyplus committed:
Merge branch 'master' of https://github.com/Strazyplus/Citadel-Station-13 into DRAKEBORG

Strazyplus committed:
Merge branch 'master' of https://github.com/Citadel-Station-13/Citadel-Station-13 into DRAKEBORG

deathride58 reviewed (approved):
it's good to merge now. thanks!
